DataFormats/L1TrackTrigger/interface/TTTrack.h

```diff
@@ -115,268 +136,232 @@ class TTTrack {
+  } else
     return 0.0;
+}
 
-  if (nPar == 4) {
-    return theChi24Par / (2 * theStubRefs.size() - 4);
-  }
-
-  if (nPar == 5) {
-    return theChi25Par / (2 * theStubRefs.size() - 5);
-  }
-
-  return 0.0;
+template <typename T>
+double TTTrack<T>::getChi2Red(unsigned int npar) const  //backwards compatibility
+{
+  return chi2Red();
 }
 
 /// StubPtConsistency
 template <typename T>
-void TTTrack<T>::setStubPtConsistency(double aStubPtConsistency, unsigned int nPar) {
-  if (!checkValidArgsForSet(nPar))
-    return;
-
-  if (nPar == 4) {
-    valid4ParFit = true;
-    theStubPtConsistency4Par = aStubPtConsistency;
-  }
-
-  if (nPar == 5) {
-    valid5ParFit = true;
-    theStubPtConsistency5Par = aStubPtConsistency;
-  }
+void TTTrack<T>::setStubPtConsistency(double aStubPtConsistency) {
+  theStubPtConsistency = aStubPtConsistency;
 
   return;
 }
 
 /// StubPtConsistency
 template <typename T>
-double TTTrack<T>::getStubPtConsistency(unsigned int nPar) const {
-  if (!checkValidArgs(nPar))
+double TTTrack<T>::getStubPtConsistency(unsigned int npar) const {
+  if (NumFitPars == 5 || NumFitPars == 4) {
+    return theStubPtConsistency;
+  } else
     return 0.0;
+}
 
-  if (nPar == 4) {
-    return theStubPtConsistency4Par;
+/// Set bits in 96-bit Track word
+template <typename T>
+void TTTrack<T>::setTrackWordBits() {
+  if (!(NumFitPars == 4 || NumFitPars == 5)) {
+    edm::LogError("TTTrack") << " setTrackWordBits method is called with NumFitPars=" << NumFitPars
+                             << " only possible values are 4/5" << std::endl;
+    return;
   }
 
-  if (nPar == 5) {
-    return theStubPtConsistency5Par;
-  }
+  unsigned int sparebits = 0;
 
-  return 0.0;
-}
+  // missing conversion of global phi to difference from sector center phi
 
-/// Duplicate identification
-template <typename T>
-bool TTTrack<T>::isTheSameAs(TTTrack<T> aTrack) const {
-  /// Take the other stubs
-  std::vector<edm::Ref<edmNew::DetSetVector<TTStub<T> >, TTStub<T> > > otherStubRefs = aTrack.getStubRefs();
-
-  /// Count shared stubs
-  unsigned int nShared = 0;
-  for (unsigned int i = 0; i < theStubRefs.size() && nShared < 2; i++) {
-    for (unsigned int j = 0; j < otherStubRefs.size() && nShared < 2; j++) {
-      if (theStubRefs.at(i) == otherStubRefs.at(j)) {
-        nShared++;
-      }
-    }
-  }
+  setTrackWord(theMomentum, thePOCA, theRInv, theChi2, theStubPtConsistency, theHitPattern, sparebits);
 
-  /// Same track if 2 shared stubs
-  return (nShared > 1);
+  return;
 }
 
-/// Quality criteria: does it have a Stub in a specific Barrel Layer?
+/// Test bits in 96-bit Track word
 template <typename T>
-bool TTTrack<T>::hasStubInBarrel(unsigned int aLayer) const {
-  for (unsigned int i = 0; i < theStubRefs.size(); i++) {
-    DetId detid = theStubRefs.at(i)->getDetId();
-    if (detid.subdetId() == StripSubdetector::TOB) {
-      return true;
-    }
-  }
+void TTTrack<T>::testTrackWordBits() {
```

Comment:
It might be better to implement this as an ostream operator (e.g. https://github.com/cms-sw/cmssw/blob/38b00fb65db80c9248afb64b78c13aadd3e86aaf/DataFormats/HcalDetId/src/HcalDetId.cc#L7-L22) or something similar, so it can work with any stream (and so `std::cout` is not called by accident in multithreaded code)

Reply:
just commented this out. It's only for my debugging anyway.